lib/options.go

```diff
@@ -44,6 +47,238 @@ const DefaultScenarioName = "default"
+func (d DNSPolicy) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// NullDNSPolicy is a nullable wrapper around DNSPolicy, required for the
+// current configuration system.
+type NullDNSPolicy struct {
+	DNSPolicy
+	Valid bool
+}
+
+// UnmarshalJSON converts JSON data to a valid NullDNSStratey
+func (d *NullDNSPolicy) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	if err := json.Unmarshal(data, &d.DNSPolicy); err != nil {
+		return err
+	}
+	d.Valid = true
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d NullDNSPolicy) MarshalJSON() ([]byte, error) {
+	if !d.Valid {
+		return []byte(`null`), nil
+	}
+	return json.Marshal(d.DNSPolicy)
+}
+
+// DNSSelect is the strategy to use when picking a single IP if more than one
+// is returned for a host name.
+//go:generate enumer -type=DNSSelect -transform=kebab -trimprefix DNS -output dns_select_gen.go
+type DNSSelect uint8
+
+const (
+	// DNSFirst returns the first IP from the response.
+	DNSFirst DNSSelect = iota + 1
+	// DNSRoundRobin rotates the IP returned on each lookup.
+	DNSRoundRobin
+	// DNSRandom returns a random IP from the response.
+	DNSRandom
+)
+
+// UnmarshalJSON converts JSON data to a valid DNSSelect
+func (d *DNSSelect) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	v, err := DNSSelectString(s)
+	if err != nil {
+		return err
+	}
+	*d = v
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d DNSSelect) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// NullDNSSelect is a nullable wrapper around DNSSelect, required for the
+// current configuration system.
+type NullDNSSelect struct {
+	DNSSelect
+	Valid bool
+}
+
+// UnmarshalJSON converts JSON data to a valid NullDNSStratey
+func (d *NullDNSSelect) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	if err := json.Unmarshal(data, &d.DNSSelect); err != nil {
+		return err
+	}
+	d.Valid = true
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d NullDNSSelect) MarshalJSON() ([]byte, error) {
+	if !d.Valid {
+		return []byte(`null`), nil
+	}
+	return json.Marshal(d.DNSSelect)
+}
+
+// DefaultDNSConfig returns the default DNS configuration.
+func DefaultDNSConfig() DNSConfig {
+	return DNSConfig{
+		TTL:    null.NewString("5m", false),
+		Select: NullDNSSelect{DNSRandom, false},
+		Policy: NullDNSPolicy{DNSpreferIPv4, false},
+	}
+}
+
+// String implements fmt.Stringer.
+func (c DNSConfig) String() string {
+	out := make([]string, 0, 2)
+	out = append(out, fmt.Sprintf("ttl=%s", c.TTL.String))
+	out = append(out, fmt.Sprintf("select=%s", c.Select.String()))
+	out = append(out, fmt.Sprintf("policy=%s", c.Policy.String()))
+	return strings.Join(out, ",")
```

Comment:
nit: Why ... just not a single fmt.Sprintf ?
Also .. the 2 should be 3 ;)

Reply:
🤦  I find this pattern more readable, but sure, I'll change it, especially to avoid the array. :)

Reply:
Fixed by 401961f3.